packages/integration-tests/src/test-e2e/create_amplify.test.ts

```diff
@@ -8,204 +8,366 @@ import assert from 'assert';
+      'registry',
+      'http://localhost:4873',
+    ]);
+    await execa(packageManagerExecutable, ['config', 'get', 'registry']);
 
-      // Force 'create-amplify' installation in npx cache by executing help command
-      // before tests run. Otherwise, installing 'create-amplify' concurrently
-      // may lead to race conditions and corrupted npx cache.
-      await execa('npm', ['create', 'amplify', '--yes', '--', '--help'], {
+    await execa(packageManagerExecutable, ['store', 'clear']);
+  }
+};
+
+void describe('create-amplify script', { concurrency }, () => {
+  const { PACKAGE_MANAGER_EXECUTABLE = 'npm' } = process.env;
+  const packageManagerExecutable =
+    PACKAGE_MANAGER_EXECUTABLE === 'yarn-stable'
+      ? 'yarn'
+      : PACKAGE_MANAGER_EXECUTABLE;
+
+  before(async () => {
+    // start a local npm proxy and publish the current codebase to the proxy
+    await execa('npm', ['run', 'clean:npm-proxy'], { stdio: 'inherit' });
+    await execa('npm', ['run', 'vend'], { stdio: 'inherit' });
+
+    // install package manager
+    if (PACKAGE_MANAGER_EXECUTABLE === 'yarn') {
+      await execa('npm', ['install', '-g', 'yarn'], { stdio: 'inherit' });
+    } else if (PACKAGE_MANAGER_EXECUTABLE === 'pnpm') {
+      await execa('npm', ['install', '-g', PACKAGE_MANAGER_EXECUTABLE], {
+        stdio: 'inherit',
+      });
+    }
+
+    // nuke the npx cache to ensure we are installing packages from the npm proxy
+    if (PACKAGE_MANAGER_EXECUTABLE !== 'yarn-stable') {
+      await packageManagerSetup(
+        PACKAGE_MANAGER_EXECUTABLE as PackageManagerExecutable
+      );
+    }
+
+    // Force 'create-amplify' installation in npx cache by executing help command
+    // before tests run. Otherwise, installing 'create-amplify' concurrently
+    // may lead to race conditions and corrupted npx cache.
+    await execa(
+      packageManagerExecutable,
+      [
+        'create',
+        'amplify',
+        ...(PACKAGE_MANAGER_EXECUTABLE === 'yarn-stable'
+          ? []
+          : ['--yes', '--']),
+        '--help',
+      ],
+      {
         // Command must run outside of 'amplify-backend' workspace.
         cwd: os.homedir(),
         stdio: 'inherit',
-      });
-    });
+      }
+    );
+  });
 
-    after(async () => {
-      // stop the npm proxy
-      await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
-    });
+  after(async () => {
+    // stop the npm proxy
+    await execa('npm', ['run', 'stop:npm-proxy'], { stdio: 'inherit' });
+  });
+
+  const initialStates = ['empty', 'module', 'commonjs'] as const;
+
+  initialStates.forEach((initialState) => {
+    void describe('installs expected packages and scaffolds expected files', () => {
+      let tempDir: string;
+      beforeEach(async () => {
+        tempDir = await fs.mkdtemp(
+          path.join(os.tmpdir(), 'test-create-amplify')
+        );
+
+        if (PACKAGE_MANAGER_EXECUTABLE === 'yarn-stable') {
+          await packageManagerSetup(
```

Comment:
`yarn-stable` should not be installed globally https://yarnpkg.com/getting-started/install